src/scripts/setup.sh

```diff
@@ -188,6 +194,45 @@ case ${config_action} in
 
 esac
 
+# Update shairport config now we know the config option
```

Comment:
Move all this to _inc/common.sh_, or better, create an _inc/shairport-sync.sh_ and put it there as a function like `shairport-sync_setup` along with the variables you have defined. Also, when there is this file, it's a good idea to move the `function install_shairport-sync_from_source` from _inc/common.sh_ into the new file. Don't forget to deal with `${TMP_DIR}` and to include this new file in _setup.sh_.

The setup script is kind of like a wizard and all the magic happens in the _inc/_ files.